### DISCUSSION TITLE
希望支持模组下载读取自定义列表配置的功能

### DISCUSSION THREAD

**Discussion by MakoStar** [Meloong-Git/PCL#1410] [提问] [answered]
## 描述
希望在下载模组窗口中添加一个支持读取自定义模组列表的配置文件(json?)功能，构想就是读取一个自定义的模组列表配置文件，然后启动器根据列表中的模组名称搜索并展示出来一个列表供下载，或者在自定义的配置文件中写好需要的模组信息(类似于CurseForge整合包那种模组配置文件)，启动器读取直接下载模组包。

## 格式
``` json
{
    "config": {
        "version": "1.19.3",
        "loader": "fabric"
    },
    "mods": [
        "mod-name-1",
        "mod-name-2",
        "mod-name-3"
    ]
}
```

**Comment by RawDiamondMC** [comment]
所以为什么不直接用CF整合包呢...
看一下#1354 是不是你想要的
（这个issue写的比较杂，建议直接看我的评论）

**Comment by 212112111zwl** [comment]
请在issue里面提交新功能提案或优化，谢谢~
当然提交前先检查有没有与已提交的issue重复……

**Reply by RawDiamondMC** [reply]
看起来是#1354 的**有效**内容

**Comment by WForst-Breeze** [comment, marked as answer]
很可惜，已有朋友与您有类似想法并提交了 #728，但是已被拒绝，感谢您的建议~

**Reply by RawDiamondMC** [reply]
那#1354 在投啥（）
那个issue写的太杂了，看不懂

**Reply by WForst-Breeze** [reply]
确实杂，那个 issue 里其实提及了好几项功能，但拐来拐去实际上考虑整的就一个：
> 在百科搜索按钮旁边加入一个按钮（这个按钮最好直接显示出来），可以直接打开该mod的pcl下载界面，或者直接打开该模组pcl的mod搜索页面，就像直接打开百科搜索页面一样。此时需要再加入一个按钮便于快速跳转回模组页面。

简单讲就是：在 **版本设置** 里的 **Mod 管理** 中加一个可以直接跳转至该 Mod 的 Mod 下载页面的按钮 ~（还是很绕）~

**Reply by RawDiamondMC** [reply]
> 确实杂，那个 issue 里其实提及了好几项功能，但拐来拐去实际上考虑整的就一个：
> 
> > 在百科搜索按钮旁边加入一个按钮（这个按钮最好直接显示出来），可以直接打开该mod的pcl下载界面，或者直接打开该模组pcl的mod搜索页面，就像直接打开百科搜索页面一样。此时需要再加入一个按钮便于快速跳转回模组页面。
> 
> 简单讲就是：在 **版本设置** 里的 **Mod 管理** 中加一个可以直接跳转至该 Mod 的 Mod 下载页面的按钮 ~（还是很绕）~

谢，差点看不懂（（（
